### PR TITLE
PHP7.1 and PSR12 compatible code

### PR DESCRIPTION
**Pull request by GeoffreyDijkstra**
Changed code so it's PSR (12) compatible (and added github actions)
Also changed composer so it requires PHP7.1 instead of PHP7.2, code was already compatible, but composer could not require package for us, also original package required PHP ^5.6, so to also make it easier for other people to switch. Side note, because this is not breaking code, you can tag it as "hotfix" or "feature" if you want.

### CODE REVIEW

**Comment by GeoffreyDijkstra** [comment]
PS: I assume (@JozefVerhoef here we go again with assuming :roll_eyes:) that you guys know that you can rebase and merge this into master so that you don't get a merge commit, also I have my fork in sync so that there are no conflicts for doing this :smile:

**Comment by GeoffreyDijkstra** [comment]
@JozefVerhoef can you merge this? Or let me know if you don't agree with something / need something changed etc :smile:

**Review by SjonHortensius — CHANGES_REQUESTED** [review]
Are the additional doc-comments in any way required by PSR12? I'm not sure if they are an improvement here

File: composer.json
Comment (pbroekhof) [inline composer.json]: Add a trailing slash to "src" i.e. "src/" is preferred.

File: composer.json
Comment (pbroekhof) [inline composer.json]: While PostcodeNl_Api_Client does replace api-restclient, it is not a drop-in replacement which "replace" would imply.

**Review by pbroekhof — CHANGES_REQUESTED** [review]

File: composer.json
Comment (GeoffreyDijkstra) [reply composer.json]: Replace in composer just means that if you also have the original package in composer that one should not be installed. But I can leave this out if you want.

File: composer.json
Comment (GeoffreyDijkstra) [reply composer.json]: This is just a matter of preference it isn't required, I'll change it.

**Comment by GeoffreyDijkstra** [comment]
@SjonHortensius it's actually required as far as I can see https://www.php-fig.org/psr/psr-12/#3-declare-statements-namespace-and-import-statements but at least codesniffer fails on it if you don't have them.

File: composer.json
Comment (pbroekhof) [reply composer.json]: As far as I know, "replace" means you tell composer that this package contains the features of the original package and version, i.e. it has its classes and can be used instead of the original package. This is mainly used when you fork a package or have a bundle of multiple packages in one package. 

Which is not the case here, so it would be best not to have it.

**Comment by pbroekhof** [comment]
> @SjonHortensius it's actually required as far as I can see https://www.php-fig.org/psr/psr-12/#3-declare-statements-namespace-and-import-statements but at least codesniffer fails on it if you don't have them.

@GeoffreyDijkstra I cannot reproduce the doc-comment requirement using the phpcs.xml ruleset in the pull request. Can you supply the codesniffer error/rule that a missing doc-comment would be triggering?

**Comment by GeoffreyDijkstra** [comment]
@pbroekhof maybe it's just me but what's the problem of having a docblocks? phpstorm generates these for me and every IDE/tool supporting dockblocks can use them. To me this point you are trying to make (which I'm missing) is not enough to block this pull request to be merged. And if you don't like something about it just checkout this branch locally make the adjustments you like and then merge it :smile: Thats the whole idea of github contributing, let somebody contribute code if it's not to your liking or standards edit it before merging (makes everybody happy) :smile:

And now to answering your question it seems this isn't a requirement anymore with latest codesniffer version (I ran codesniffer again on this project, also found that example directory was still checked so fixed that)

**Comment by pbroekhof** [comment]
@GeoffreyDijkstra I don't mind the docblocks myself (though they can be a bit superfluous when empty), I was mainly concerned why there was a difference in PSR12 codesniffer warnings between your checkout and my own, due to your earlier comment. I wanted to resolve that before accepting the codesniffer PSR12 integration. So it wasn't really the docblocks themselves that were blocking :)

In any case, I prefer to (where possible) let the requester answer/change/tweak things first before considering taking over the pull request myself. But I think we've resolved all blocking issues now, thanks for the effort!

**Review by pbroekhof — APPROVED** [review]

**Pull request merged**
